dts/arm/ti/am62x_sk_m4.dtsi

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2023 Texas Instruments Incorporated
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <mem.h>
+#include <freq.h>
+#include <arm/armv7-m.dtsi>
+#include <zephyr/dt-bindings/pinctrl/ti-k3-pinctrl.h>
+
+/ {
+
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu0: cpu@0 {
+			device_type = "cpu";
+			compatible = "arm,cortex-m4f";
+			reg = <0>;
+		};
+	};
+
+	sram0: memory@0 {
+		compatible = "mmio-sram";
+		reg = <0x0 DT_SIZE_K(192)>;		/* 192 KB of SRAM (I-Code) */
+	};
+
+	sram1: memory1@40000 {
+		compatible = "mmio-sram";
+		reg = <0x40000 DT_SIZE_K(64)>;		/* 64 KB of SRAM (D-Code) */
+	};
+
+	sysclk: system-clock {
+		compatible = "fixed-clock";
+		clock-frequency = <400000000>;
+		#clock-cells = <0>;
+	};
+
+	pinctrl: pinctrl@4084000 {
+		compatible = "ti,k3-pinctrl";
+		reg = <0x04084000 0x88>;
+		status = "okay";
+	};
+
+	uart0: serial@4a00000 {
+		compatible = "ns16550";
+		reg = <0x04a00000 0x200>;
+		interrupts = <40 4>;
```

Comment:
The interrupt should be 24:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/124088381/f03da761-5b51-4be3-81dd-be2d05c5ba05)
I tested this with `interrupts = <24 4>;` and the shell module sample with ISR driver UARTs works!

Reply:
BTW thanks for fixing this properly, rather than the quick fix (disabling UART ISRs) I had suggested

Reply:
Fixed this. Thank you for the review!